src/App.js

```diff
@@ -1,26 +1,231 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
 
-function App() {
+function Display(props) {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div> 
+    {props.value}
     </div>
+    );
+}
+
+function Button(props) {
+
+  return (
+  <button
+    onClick= {() => props.onClick(props.value)}> 
+  {props.value}
+  </button>
   );
 }
 
-export default App;
+class Calculator extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      previousValue: "0",
+      displayValue: "0",
+      operator: "",
+      valueIsDecimal: false,
+      valueIsNegative: false,
+    }
+  }
+
+  handleClick(i){
+    const numbers = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
+    const operators = ["+", "-", "x", "÷"];
+
+    if(numbers.includes(i)){
```

Comment:
you have a big nesting just because of this if. I guess you could just do:

`if(!numbers.includes(i)) { return; }`

and avoid the nesting